pkg/plugins/k8s/k8s_plugin.go

```diff
@@ -0,0 +1,350 @@
+package main
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path"
+	"strings"
+
+	"github.com/coreos/go-systemd/v22/unit"
+	"github.com/golang/glog"
+
+	sriovnetworkv1 "github.com/k8snetworkplumbingwg/sriov-network-operator/api/v1"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/service"
+	"github.com/k8snetworkplumbingwg/sriov-network-operator/pkg/utils"
+)
+
+type K8sPlugin struct {
+	PluginName            string
+	SpecVersion           string
+	serviceManager        service.ServiceManager
+	switchdevRunScript    *service.ScriptManifestFile
+	switchdevService      *service.Service
+	openVSwitchService    *service.Service
+	networkManagerService *service.Service
+	updateTarget          *k8sUpdateTarget
+}
+
+type k8sUpdateTarget struct {
+	switchdevService bool
+	switchdevScript  bool
+	systemServices   []*service.Service
+}
+
+func (u *k8sUpdateTarget) needUpdate() bool {
+	return u.switchdevService || u.switchdevScript || len(u.systemServices) > 0
+}
+
+func (u *k8sUpdateTarget) reset() {
+	u.switchdevService = false
+	u.switchdevScript = false
+	u.systemServices = []*service.Service{}
+}
+
+func (u *k8sUpdateTarget) String() string {
```

Comment:
where are you using it? and why is with capital latter?

Reply:
On line 139 in print the log message, function `String` is used to represent any object as a string in go